packages/x-data-grid/src/utils/createSelector.ts

```diff
@@ -18,20 +18,12 @@ type GridCreateSelectorFunction = ReturnType<typeof reselectCreateSelector> & {
   selectorArgs?: any;
 };
 
-// TODO v8: Remove this type
-export interface OutputSelector<State, Result> {
-  (apiRef: React.MutableRefObject<{ state: State; instanceId: GridCoreApi['instanceId'] }>): Result;
-  (state: State, instanceId: GridCoreApi['instanceId']): Result;
-  acceptsApiRef: boolean;
-}
-
-// TODO v8: Rename this type to `OutputSelector`
-export interface OutputSelectorV8<State, Args, Result> {
+export interface OutputSelector<State, Args, Result> {
   (
     apiRef: React.MutableRefObject<{ state: State; instanceId: GridCoreApi['instanceId'] }>,
     args?: Args,
   ): Result;
-  (state: State, instanceId: GridCoreApi['instanceId']): Result;
+  (state: State, args?: Args, instanceId?: GridCoreApi['instanceId']): Result;
```

Comment:
Instance ID parameter being required feels unnecessary.
1. We allow slicing data without passing it, it could be fine if the user is aware of the performance issues a respective selector may cause.
2. It doesn't apply to non-memoized selectors.
3. Even for the memoized selectors, we warn the users to beware of the possible performance issues and act accordingly

With https://github.com/mui/mui-x/issues/11440 it would even be more relevant as the only officially supported way would be the `selector(apiRef)`. Comments are welcome.

Reply:
As long as we do #11440, we can get rid of those for v8.